src/main/java/com/cloudbees/jenkins/plugins/bitbucket/endpoints/AbstractBitbucketEndpoint.java

```diff
@@ -112,15 +115,15 @@ public final String getCredentialsId() {
     }
 
     /**
-     * Looks up the {@link StandardUsernamePasswordCredentials} to use for auto-management of hooks.
+     * Looks up the {@link StandardCredentials} to use for auto-management of hooks.
      *
      * @return the credentials or {@code null}.
      */
     @CheckForNull
-    public StandardUsernamePasswordCredentials credentials() {
+    public StandardCredentials credentials() {
         return StringUtils.isBlank(credentialsId) ? null : CredentialsMatchers.firstOrNull(
                 CredentialsProvider.lookupCredentials(
-                        StandardUsernamePasswordCredentials.class,
+                        StandardCredentials.class,
                         Jenkins.getActiveInstance(),
                         ACL.SYSTEM,
                         URIRequirementBuilder.fromUri(getServerUrl()).build()
```

Comment:
You will also need to use the `AuthenticationTokens` matcher. (Further, it may be better to use the "context" variant... so we can decorate with the URL scheme and restrict which Authentication token types are valid against different URLs... i.e. you cannot have client certificates for `http` but only for `https` but that can be handled later as a finesse)

Reply:
CredentialsMatchers are used below so all looks good